compiler/extra/gbc-check.py

```diff
@@ -0,0 +1,94 @@
+
+# Simple script to test easily multiple bond files and compare the
+# output of two versions of gbc. Output includes all codegen output,
+# stdout and stderr. Also compares quantity of codegen files generated
+# and exit code. A diff will be outputed per file.
+
+import argparse
+import difflib
+import glob
+import os
+import subprocess
+import shutil
+
+def get_bond_files(base_dir):
+    bond_files = []
+
+    os.chdir(base_dir)
+    for file in glob.glob("**/*.bond", recursive=True):
+        bond_files.append(os.path.join(base_dir, file))
+
+    print("INFO Found {} bond files.".format(len(bond_files)))
+    return bond_files
+
+def generate_bond(gbc, gbc_flags, output_dir, bond_file):
+    bond_command = "{} {} -o {} {}".format(gbc, gbc_flags, output_dir, bond_file)
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    with open(os.path.join(output_dir,"stdout.txt"), "w+") as stdout_file:
+        with open(os.path.join(output_dir,"stderr.txt"), "w+") as stderr_file:
+            return_code = subprocess.call(bond_command, shell=True, stdout=stdout_file, stderr=stderr_file)
```

Comment:
It looks like the following will also work and will be more robust to things like spaces in paths:
```
subprocess.call(gbc, gbc_flags, "-o", output_dir, bond_file, shell=False, stdout=stdout_file, stderr=stderr_file)
```